Setup.hs

```diff
@@ -0,0 +1,77 @@
+-- See the comments for 'ppProto'.
+
+{-# OPTIONS_GHC -Wall #-}
+
+import Distribution.Simple
+         (UserHooks(..), defaultMainWithHooks, simpleUserHooks)
+import Distribution.Simple.PreProcess (PreProcessor(..))
+import Distribution.Simple.Program (runProgram)
+import Distribution.Simple.Program.Db (requireProgram)
+import Distribution.Simple.Program.Types (Program(..), simpleProgram)
+import Distribution.Simple.Utils (findProgramVersion, info)
+import Distribution.Types.BuildInfo (BuildInfo)
+import Distribution.Types.LocalBuildInfo (LocalBuildInfo(..))
+import Distribution.Types.ComponentLocalBuildInfo (ComponentLocalBuildInfo)
+import Distribution.Verbosity (Verbosity)
+import System.FilePath ((</>), normalise)
+
+main :: IO ()
+main = defaultMainWithHooks customHooks
+
+customHooks :: UserHooks
+customHooks = simpleUserHooks
+  { hookedPreProcessors =
+      hookedPreProcessors simpleUserHooks ++ [( "proto", ppProto )]
```

Comment:
Nit: Why not prepend your custom preprocessor to the list instead of append it?

Reply:
Sure; changed in 89c11c99d9f5bc62dce2fdcb542740ed5f8ece1b .